src/Site.js

```diff
@@ -385,6 +412,193 @@ Site.prototype.createPage = function (config) {
+      .then(resolve)
+      .catch(reject);
+  });
+};
+
+/**
+ * Builds a site navigation file from the directory structure of the site.
+ * @param rootPath
+ * @param siteLayoutSiteNavDefaultPath
+ */
+Site.prototype.buildSiteNav = function (rootPath, siteLayoutSiteNavDefaultPath) {
+  return new Promise((resolve, reject) => {
+    this.readSiteConfig()
+      .then(() => this.collectAddressablePages())
+      .then(() => {
+        let siteNavContent = '';
+        this.addressablePages
+          .filter(addressablePage => !addressablePage.src.startsWith('_'))
+          .forEach((page) => {
+            const addressablePagePath = path.join(rootPath, page.src);
+            const relativePagePathWithoutExt = FsUtil.removeExtension(
+              path.relative(rootPath, addressablePagePath));
+            const pageName = _.startCase(FsUtil.removeExtension(path.basename(addressablePagePath)));
+            const pageUrl = `{{ baseUrl }}/${relativePagePathWithoutExt}.html`;
+            siteNavContent += `* [${pageName}](${pageUrl})\n`;
+          });
+        const wrappedSiteNavContent = `<navigation>\n${siteNavContent}\n</navigation>`;
+        return fs.outputFileAsync(siteLayoutSiteNavDefaultPath, wrappedSiteNavContent);
+      })
+      .then(resolve)
+      .catch(reject);
+  });
+};
+
+/**
+ * Applies the default layout to all addressable pages by modifying the site config file.
+ * @param rootPath
+ */
+Site.prototype.addDefaultLayoutToSiteConfig = function (rootPath) {
+  const configPath = path.join(rootPath, SITE_CONFIG_NAME);
+
+  return new Promise((resolve, reject) => {
+    fs.readJsonAsync(configPath)
+      .then((config) => {
+        const layoutObj = { glob: '**/*.+(md|mbd)', layout: LAYOUT_DEFAULT_NAME };
+        config.pages.push(layoutObj);
+        return fs.outputJsonAsync(configPath, config);
+      })
+      .then(resolve)
+      .catch(reject);
+  });
+};
+
+/**
+ * Adds top navigation to site and includes it in all addressable pages.
+ * @param rootPath
+ */
+Site.prototype.addTopNavToAddressablePages = function (rootPath) {
+  const topNavDefaultPath = path.join(rootPath, TOP_NAV_PATH);
+  return new Promise((resolve, reject) => {
+    fs.accessAsync(topNavDefaultPath)
+      .catch(() => {
+        if (fs.existsSync(topNavDefaultPath)) {
+          return Promise.resolve();
+        }
+        return fs.outputFileAsync(topNavDefaultPath, TOP_NAV_DEFAULT);
+      })
+      .then(() => this.readSiteConfig())
+      .then(() => this.collectAddressablePages())
+      .then(() => {
+        this.addressablePages
+          .filter(addressablePage => !addressablePage.src.startsWith('_'))
+          .forEach((page) => {
+            const addressablePagePath = path.join(rootPath, page.src);
+            const topNavRelativePath = path.relative(rootPath, topNavDefaultPath);
+            if (topNavRelativePath.length > 0) {
+              const includeTopNavCode = `<include src="${topNavRelativePath}" />\n\n`;
```

Comment:
This seems quite expensive, i'm not too sure but is there a way to to specify this in layouts/front matter?

Reply:
As of now, there is no support for a `header.md` like we have with footers. 
I think I would probably need to use the functionality in #733 once it's added...